packages/kit/src/exports/vite/graph_analysis/index.js

```diff
@@ -13,9 +15,21 @@ const ILLEGAL_MODULE_NAME_PATTERN = /.*\.server\..+/;
  *   cwd: string;
  *   node_modules: string;
  *   server: string;
+ *   svelte_config: (string|RegExp|((filename: string)=>undefined|boolean))[];
  * }} dirs
  */
 export function is_illegal(id, dirs) {
+	//svelte_config is processed as first, so the function rule can make exceptions
+	for (const configRule of dirs.svelte_config) {
+		if (typeof configRule === 'string') {
+			if (id.startsWith(configRule)) return true;
+		} else if (configRule instanceof RegExp) {
+			if (configRule.test(id)) return true;
+		} else if (typeof configRule === 'function') {
+			const check = configRule(id);
+			if (typeof check === 'boolean') return check; //always return the boolean to allow exceptions, if its undefined continue to allow multiple rules
```

Comment:
early return should only happen on true, otherwise a return false could give access to an id that would be denied by a later check

Reply:
that would be a design decision. If I want to make a exception and allow a file import, no matter what other rules say, the early return is needed. but depends on execution order then. If a rule dont care, it returns undefined/nothing

Reply:
if really reducing to only one rule and not allowing exceptions i'd drop anything but regex, should be flexible enough IMO